test/lib/dashboard-agent.spec.js

```diff
@@ -64,18 +66,38 @@ describe("dashboard-agent", function () {
 
       var checkMetrics = function (metrics) {
         expect(metrics).to.be.an("object");
-        expect(metrics.eventLoop).to.deep.equal({ delay: 0, high: 0 });
-        expect(metrics.mem.systemTotal).to.be.above(0);
-        expect(metrics.mem.rss).to.be.above(0);
-        expect(metrics.mem.heapTotal).to.be.above(0);
-        expect(metrics.mem.heapUsed).to.be.above(0);
-        expect(metrics.cpu.utilization).to.be.above(0);
+        expect(metrics.eventLoop.delay).to.be.a("number");
+        expect(metrics.eventLoop.high).to.be.a("number");
+        expect(metrics.mem.systemTotal).to.equal(20);
+        expect(metrics.mem.rss).to.equal(30);
+        expect(metrics.mem.heapTotal).to.equal(40);
+        expect(metrics.mem.heapUsed).to.equal(50);
+        expect(metrics.cpu.utilization).to.equal(60);
       };
 
+      var stubProcess = sinon.stub(process, "memoryUsage", function () {
```

Comment:
GLOBAL: Never use `sinon.stub|spy` on anything that requires a `.restore()`. Instead, use a `sandbox` which we declare at top scope for this test suite. 

Here you've got a problem that if anything fails / errors in this test, you won't reach the `.restore()`s on L99-L100. That means you have real methods potentially left unrestored that could cause unrelated tests to fail. Thus, you must always try/catch synchronous stubs to restore _or_ do it in a `beforeEach|afterEach` pair.

Since we already do this with a `sandbox`, you can just _use_ that to `sandbox.stub` and never worry about the restore, since a global restore is done in the top-level `afterEach`